### PR TITLE
Close feedback survey after picking an option

### PR DESCRIPTION
**Pull request by vegaro**
Cleans up the flow a bit, so the feedback survey is closed after picking an option and doesn't stay there after cancelling a subscription

This is needed to be able to send the selected feedback survey option to the backend as event.

### CODE REVIEW

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift [hunk 1]
```diff
@@ -59,8 +85,21 @@ struct FeedbackSurveyView: View {
...
+            NavigationLink(destination: EmptyView(), isActive: $isOptionSelected) {
+                EmptyView()
+            }
+            .hidden()
+        )
     }
 
+    private func handleOptionSelected() {
+        if #available(iOS 16.0, *), let navigationPath = navigationPathBinding {
+            navigationPath.wrappedValue.removeLast()
```
Comment (vegaro) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:98]: @fire-at-will this is not working, by the time it gets here, the stack is empty

File: RevenueCatUI/CustomerCenter/Views/CustomerCenterView.swift [hunk 2]
```diff
@@ -64,6 +78,11 @@ public struct CustomerCenterView: View {
             await loadInformationIfNeeded()
         }
         .environmentObject(self.viewModel)
+        .onAppear {
```
Comment (fire-at-will) [inline RevenueCatUI/CustomerCenter/Views/CustomerCenterView.swift:81]: Nit: It might be simpler to do this initialization where `_navigationPathWrapper` is declared instead of here in an onAppear. Something like:

```swift
@State private var _navigationPathWrapper: Any? = {
    if _navigationPathWrapper == nil, #available(iOS 16.0, *) {
                return NavigationPathWrapper()
    } else {
        return nil
   }
}()
```

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift [hunk 3]
```diff
@@ -59,8 +85,21 @@ struct FeedbackSurveyView: View {
         }
         .navigationTitle(self.viewModel.feedbackSurveyData.configuration.title)
         .navigationBarTitleDisplayMode(.inline)
+        .background(
+            NavigationLink(destination: EmptyView(), isActive: $isOptionSelected) {
+                EmptyView()
+            }
+            .hidden()
+        )
```
Comment (fire-at-will) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:93]: Why is this included?

File: RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift [hunk 4]
```diff
@@ -50,26 +69,32 @@ struct ManageSubscriptionsView: View {
...
-            content
-                .navigationDestination(isPresented: .isNotNil(self.$viewModel.feedbackSurveyData)) {
-                    if let feedbackSurveyData = self.viewModel.feedbackSurveyData {
-                        FeedbackSurveyView(feedbackSurveyData: feedbackSurveyData,
-                                           customerCenterActionHandler: self.customerCenterActionHandler)
+        content
+            .background(
+                Group {
+                    if #available(iOS 16.0, *), let navigationPath = navigationPathBinding {
+                        content
```
Comment (fire-at-will) [inline RevenueCatUI/CustomerCenter/Views/ManageSubscriptionsView.swift:76]: Is it necessary to re-render `content` in the background on `content`?

It might be more efficient to place these modifiers on something that we know won't have a lot of rendering complexity. I'll often use something like `Color.clear` for things like this, and set the opacity to 0 to be safe

**Comment by vegaro** [comment]
@fire-at-will I was struggling so much to make this work with `NavigationStack`... I tried another simpler approach. I think we need to rethink the way we do navigation. We've already discussed that in the past, and I think it is going to be one of the next things to address. We probably need to build our own system

Anyways, I think this works for now and it's now ready for review

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift [hunk 5]
```diff
@@ -33,19 +33,27 @@ struct FeedbackSurveyView: View {
     @Environment(\.colorScheme)
     private var colorScheme
 
+    @Binding
+    private var isPresented: Bool
```
Comment (tonidero) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:37]: Hmm I'm a bit confused, this seems to only be assigned but not read?

Comment (vegaro) [reply RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:37]: this might be a bit confusing yes... I create a binding here https://github.com/RevenueCat/purchases-ios/pull/4444/files#diff-121dcc7927012032acdbbf026716bf9daf66d91cee530ab12eae052d06526aa0R68

```
.isNotNil(self.$viewModel.feedbackSurveyData)
```

So if I set that to false, it nullifies the `feedbackSurveyData` which hides the feedback survey, since it's only presented if `feedbackSurveyData` is not nil.

Another approach I can use is adding an `onDismiss` callback:

```
.navigationDestination(isPresented: .isNotNil(self.$viewModel.feedbackSurveyData)) {
      if let feedbackSurveyData = self.viewModel.feedbackSurveyData {
          FeedbackSurveyView(feedbackSurveyData: feedbackSurveyData,
                             customerCenterActionHandler: self.customerCenterActionHandler,
                             onDismiss: { self.viewModel.feedbackSurveyData = nil })
      }
  }
```

Maybe that's more clear?

Comment (tonidero) [reply RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:37]: Ahhh got it! Well, I think it might be clearer, but I think it's ok as is. Thanks!

**Review by tonidero — APPROVED** [review]
